### PR TITLE
DAOS-7204 EC: a few fixes for EC agg

### PR DESCRIPTION
**Pull request by liuxuezhao**
1. agg_reset_pos() after yield during vos_iterate().
2. fix a bug in cont_ec_eph_reduce() that possibly cause issue for
   agg epoch boundary, and then cause vos agg before ec agg.
3. fix a issue in daos_aggregate_ec.c, sleep 40 second cannot ensure
   agg happens, enlarge to 45 S.
4. A few other changes for logs.

Signed-off-by: Xuezhao Liu <xuezhao.liu@intel.com>

### CODE REVIEW

File: src/container/srv_target.c [hunk 1]
```diff
@@ -2324,10 +2324,16 @@ cont_ec_eph_reduce(void *agg_arg, void *xs_arg)
...
-		     x_arg->ephs[i].eph > c_eph->ce_last_eph))
+		if (c_eph->ce_eph == 0) {
 			c_eph->ce_eph = x_arg->ephs[i].eph;
+			continue;
+		}
+		if (x_arg->ephs[i].eph != 0 &&
+		    x_arg->ephs[i].eph > c_eph->ce_last_eph) {
+			if (c_eph->ce_eph <= c_eph->ce_last_eph ||
+			    x_arg->ephs[i].eph < c_eph->ce_eph)
+				c_eph->ce_eph = x_arg->ephs[i].eph;
```
Comment (NiuYawei) [inline src/container/srv_target.c:2335]: This is hard to follow and I think it's still not quite right, why can't we just set the 'ce_eph' as the smallest one of the 'ephs'?

Comment (liuxuezhao) [reply src/container/srv_target.c]: refreshed, not sure if some reason need to be written as original way, will confirm with @wangdi1

File: src/container/srv_target.c [hunk 2]
```diff
@@ -2324,10 +2325,12 @@ cont_ec_eph_reduce(void *agg_arg, void *xs_arg)
...
 						  x_arg->ephs[i].cont_uuid);
-		if (c_eph->ce_eph == 0 ||
-		    (x_arg->ephs[i].eph != 0 &&
-		     x_arg->ephs[i].eph > c_eph->ce_last_eph))
+		if (c_eph->ce_first) {
 			c_eph->ce_eph = x_arg->ephs[i].eph;
+			c_eph->ce_first = 0;
+		} else if (x_arg->ephs[i].eph < c_eph->ce_eph) {
+			c_eph->ce_eph = x_arg->ephs[i].eph;
+		}
```
Comment (NiuYawei) [inline src/container/srv_target.c:2333]: Minor: The 'ce_first' looks bit confusing to me, why can't we just find out the smallest one and assign it to ce_eph?

BTW,  can we make sure it works well if any target/rank is in downout? (the eph on the target/rank never bumped)

**Review by NiuYawei — APPROVED** [review]

File: src/container/srv_target.c (hunk 2 above)
Comment (liuxuezhao) [reply src/container/srv_target.c:2333]: 'ce_first' is for the case that originally the ce_eph is zero, and no one "x_arg->ephs[i].eph" can < it. So it need to assign one value for the first time.

For "can we make sure it works well if any target/rank is in downout? (the eph on the target/rank never bumped)" seems we need some similar logic as in "ds_cont_tgt_open(), ds_pool_get_failed_tgt_idx" for the collective. I can update in this patch if it need refresh later, or in a following patch.

File: src/tests/suite/daos_aggregate_ec.c [hunk 3]
```diff
@@ -785,7 +785,9 @@ test_all_ec_agg(void **statep)
 	test_half_stripe(&ctx);
 	test_partial_stripe(&ctx);
 	test_range_punch(&ctx);
-	sleep(40);
+	print_message("sleep 45 seconds for aggregation ...\n");
+	sleep(45);
+	print_message("verification after aggregation\n");
```
Comment (wangdi1) [inline src/tests/suite/daos_aggregate_ec.c:790]: probably need set pool property to "time" and use force aggregation to make sure the aggregation really happen here.

**Review by wangdi1 — DISMISSED** [review]

File: src/tests/suite/daos_aggregate_ec.c (hunk 3 above)
Comment (liuxuezhao) [reply src/tests/suite/daos_aggregate_ec.c:790]: ok, sure.

File: src/container/srv_target.c (hunk 2 above)
Comment (liuxuezhao) [reply src/container/srv_target.c:2333]: the suggested code seems not correct for some cases like:
eph array: [0]- 11, [1] 0, [2] 12
it will incorrectly set the result as 12

Comment (wangdi1) [reply src/container/srv_target.c:2333]: hmm, it should be 0,   why 12?

**Review by wangdi1 — DISMISSED** [review]

**Review by wangdi1 — APPROVED** [review]

**Review by NiuYawei — APPROVED** [review]
